### ISSUE TITLE
Return multidevice replicated tensors to host without N_DEVICESx memory overhead.

### ISSUE THREAD

**Issue by jameszianxuTT** [tenstorrent/tt-mlir#8573]
Kind of the opposite problem to https://github.com/tenstorrent/tt-xla/issues/4011 - this time, returning tensors to host on large device-count machines is quite inefficient.

On any multidevice (single or multihost) inference workload, we return results to host _very_ inefficiently, resulting in a host memory spike scaling with wasted allocations on order of  N_DEVICES.

This shows up in kimi k2 inference, where batch 64 logits are size 64x1x163840xbf16 and _already_ all-gathered on each device (which also wastes device DRAM, but that's a separate issue).

Alone, this tensor is ~20M  in size. 

When we return this to host in runtime with toHostSingleTensor, we call ttnn from_device -> tensor.cpu -> metal enqueue_read_tensor  which allocates num_devices x tensor size return buffers and pulls the same replicated shard from every result tensor back to device, then from tt-xla, we discard all but the first if we know the output is replicated.

This problem doubles in magnitude actually, because toHostSingleTensor also untilizes all these redundant tensors on host for blackhole and some datatypes. Untilization results in an additional output buffer of the same size created on host to store the untilized output.

<img width="743" height="335" alt="Image" src="https://github.com/user-attachments/assets/30fff2ea-af2b-4406-b23b-84bbcf9ac150" />

There are two solutions I can think of here:
1. Shard the output, and create a more efficient way of returning tensors to host that's not all gathering them on device, then all gathering them again to host and discarding all but one shard. This will require changes to torch-xla probably, to intercept the x+=0 minigraph and replace it with a custom runtime call to iterate over shards and move them to host, then gather them on host.  This is efficient in device memory and host memory, but requires changes to torch-xla; or some circumvention route. 
2. Gather the output on device, and transfer one shard back to host. This will require inventing high level single shard read or linking lower level to other metal APIs, and a different tt-mlir API as well

edit: Thanks to @mstojkovicTT solution 1 is implemented - https://github.com/tenstorrent/pytorch-xla/pull/35
